x/tradeshield/keeper/msg_cancel_spot-orders.go

```diff
@@ -0,0 +1,34 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/elys-network/elys/x/tradeshield/types"
+)
+
+func (k msgServer) CancelSpotOrders(goCtx context.Context, msg *types.MsgCancelSpotOrders) (*types.MsgCancelSpotOrdersResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if len(msg.SpotOrderIds) == 0 {
+		return nil, types.ErrSizeZero
+	}
+
+	// loop through the spot orders and execute them
+	for _, spotOrderId := range msg.SpotOrderIds {
+		// get the spot order
+		spotOrder, found := k.GetPendingSpotOrder(ctx, spotOrderId)
+		if !found {
+			return nil, types.ErrSpotOrderNotFound
+		}
+
+		if spotOrder.OwnerAddress != msg.Creator {
+			return nil, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "incorrect owner")
+		}
+		
+		k.RemovePendingSpotOrder(ctx, spotOrderId)
+	}
+
+	return &types.MsgCancelSpotOrdersResponse{}, nil
```

Comment:
@cryptokage1996 we should emit an event with the new status of the order "canceled" so that an indexing service can capture the order status change as we are remove the order from the chain state. cc @Wgil